demos/ext/rust_magnus_geo/src/lib.rs

```diff
@@ -77,11 +77,14 @@ impl Rectangle {
     }
 
     /// Checks if the given `Point` is inside this `Rectangle`.
-    fn contains(&self, _other: &Point) -> Result<bool, Error> {
-        let _top_left = self.top_left.try_convert::<&Point>()?;
-        let _bottom_right = self.bottom_right.try_convert::<&Point>()?;
+    fn contains(&self, other: &Point) -> Result<bool, Error> {
+        let top_left = self.top_left.try_convert::<&Point>()?;
+        let bottom_right = self.bottom_right.try_convert::<&Point>()?;
 
-        Err(Error::new(not_imp_error(), "finish me!"))
+        Ok(   other.x >= top_left.x
```

Comment:
Small tip: check out the rust-analyzer vscode extension. It will setup auto-formatting for ya

Reply:
i used vim, and apparently my setup (probably ale) brings a free linter that removed the unused `not_imp_error` but i don't have auto-formatting setup